packages/flutterfire_ui/lib/src/auth/screens/email_verification_screen.dart

```diff
@@ -0,0 +1,176 @@
+  final List<FlutterFireUIAction> actions;
+  final double? headerMaxExtent;
+  final SideBuilder? sideBuilder;
+  final TextDirection? desktoplayoutDirection;
+  final double breakpoint;
+  final ActionCodeSettings? actionCodeSettings;
+
+  const EmailVerificationScreen({
+    Key? key,
+    this.auth,
+    this.actions = const [],
+    this.headerBuilder,
+    this.headerMaxExtent,
+    this.sideBuilder,
+    this.desktoplayoutDirection,
+    this.breakpoint = 500,
+    this.actionCodeSettings,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return FlutterFireUIActions(
+      actions: actions,
+      child: UniversalScaffold(
+        body: ResponsivePage(
+          breakpoint: breakpoint,
+          desktopLayoutDirection: desktoplayoutDirection,
+          headerBuilder: headerBuilder,
+          headerMaxExtent: headerMaxExtent,
+          sideBuilder: sideBuilder,
+          maxWidth: 1200,
+          contentFlex: 2,
+          child: Padding(
+            padding: const EdgeInsets.all(32),
+            child: _EmailVerificationScreenContent(
+              auth: auth,
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _EmailVerificationScreenContent extends StatefulWidget {
+  final FirebaseAuth? auth;
+  final ActionCodeSettings? actionCodeSettings;
+
+  const _EmailVerificationScreenContent({
+    Key? key,
+    this.auth,
+    this.actionCodeSettings,
+  }) : super(key: key);
+
+  @override
+  State<_EmailVerificationScreenContent> createState() =>
+      __EmailVerificationScreenContentState();
+}
+
+class __EmailVerificationScreenContentState
+    extends State<_EmailVerificationScreenContent> {
+  late final service = EmailVerificationService(auth);
+  FirebaseAuth get auth => widget.auth ?? FirebaseAuth.instance;
+  User get user => auth.currentUser!;
+  bool isLoading = false;
+
+  @override
+  void initState() {
+    SchedulerBinding.instance!.addPostFrameCallback(_sendEmailVerification);
+    super.initState();
+  }
+
+  void _sendEmailVerification(_) {
+    service
+      ..addListener(() {
+        setState(() {});
+
+        if (state == EmailVerificationState.verified) {
+          final action = FlutterFireUIAction.ofType<EmailVerified>(context);
+          action?.callback();
+        }
+      })
+      ..sendVerificationEmail(
+        Theme.of(context).platform,
+        widget.actionCodeSettings,
+      );
+  }
+
+  EmailVerificationState get state => service.state;
+
+  @override
+  Widget build(BuildContext context) {
+    final l = FlutterFireUILocalizations.labelsOf(context);
+
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        const Align(
+          child: Title(text: 'Verify your email'),
+        ),
+        const SizedBox(height: 32),
+        const Text(
+          'A verification email has been sent to your email address. '
```

Comment:
Guessing this and other texts should be localized in that future PR right?

Reply:
https://github.com/firebase/flutterfire/pull/8358#discussion_r837589464